sympy/series/approximants.py

```diff
@@ -101,3 +108,283 @@ def approximants(l, X=Symbol('x'), simplify=False):
+        returns `p(x), q(x)`, where `p(x)/q(x)` is the pade approximant
+
+    Examples
+    ========
+
+    >>> import sympy as sp
+    >>> from sympy.series.approximants import pade_approximant_gcdex
+    >>> x = sp.symbols('x')
+
+    >>> exp_taylor_poly = sp.Poly(x**4/24 + x**3/6 + x**2/2 + x + 1, x, domain='QQ')
+    >>> num_exp, denom_exp = pade_approximant_gcdex(exp_taylor_poly, 2)
+    >>> print(num_exp, denom_exp)
+    Poly(1/4*x**2 + 3/2*x + 3, x, domain='QQ') Poly(1/4*x**2 - 3/2*x + 3, x, domain='QQ')
+
+    The numerators and denominators of an `[m/n]` pade approximant do not
+    necessarily have order `m` and `n` respectively.
+
+    >>> sin_taylor_poly = sp.Poly(x**5/120 - x**3/6 + x, x, domain='QQ')
+    >>> num_sin, denom_sin = pade_approximant_gcdex(sin_taylor_poly, 1, 3)
+    >>> print(num_sin, denom_sin)
+    Poly(36*x, x, domain='QQ') Poly(6*x**2 + 36, x, domain='QQ')
+
+    The `[m/0]` pade approximant is the same a truncating to order `m`
+
+    >>> poly = sp.Poly(4*x**4 + 3*x**3 + 2*x**2 + x + 1, x, domain='QQ')
+    >>> num_cos, denom_cos = pade_approximant_gcdex(poly, 3, 0)
+    >>> print(num_cos, denom_cos)
+    Poly(3*x**3 + 2*x**2 + x + 1, x, domain='QQ') Poly(1, x, domain='QQ')
+
+    The pade approximant does not always exist, for example, the `[1/1]` approximant for `cos(x)`
+
+    >>> cos_taylor_poly = sp.Poly(1 - x**2/2, x, domain='QQ')
+    >>> num_cos11, denom_cos11 = pade_approximant_gcdex(cos_taylor_poly, 1)
+    >>> print(num_cos11, denom_cos11)
+    Poly(2*x, x, domain='QQ') Poly(2*x, x, domain='QQ')
+
+    `num_cos11/denom_cos11 == 1`, and `1` does not match the taylor expansion of `cos(x)`
+    to second order in `x`.
+
+    See also
+    ========
+
+    sympy.series.approximants.approximants
+    sympy.series.approximants.pade_approximant
+    sympy.series.approximants.pade_approximants_gcdex
+    """
+    if n is None:
+        n = m
+
+    if m < 0 or n < 0:
+        raise ValueError("m and n must be non-negative integers")
+
+    if not f.is_univariate:
+        raise ValueError("f must be a univariate polynomial")
+
+    min_degree = f.EM()[0]
+    if min_degree > m:
+        raise ValueError(f"polynomial has zero of order {min_degree}, " \
+        f"which is greater than {m}, the requested order of the numerator.")
+
+    approximants = pade_approximants_gcdex(f, m + n)
+
+    numerator, denominator = next(approximants)
+    for next_numerator, next_denominator in approximants:
+        if next_numerator.degree() < m:
+            return numerator, denominator
+
+        numerator, denominator = next_numerator, next_denominator
+
+    if numerator.degree() == m:
+        return numerator, denominator
+
+    # if it hasn't returned yet, there is no [m/n] pade approximant
+    return None, None
+
+
+@public
+def pade_approximant(
+    f: Expr, x: Expr, m: int, n: int | None = None
+    ) -> Expr | None:
+    """
+    `[m/n]` pade approximant of `f` around `x=0`.
```

Comment:
This signature would make it awkward to add an `x0` parameter in future. Maybe that parameter should be added now like for Mathematica's PadeApproximant function.